File: 프로그래머스/python/LV2/영어 끝말잇기.py
def solution(n, words):
    answer = []
    used = []
    for i in range(len(words)):
        if i > 0 and used[-1][-1] != words[i][0]:
            answer.append(i % n + 1)
            answer.append(((i + 1 - answer[0]) // n) + 1)
            break

        if words[i] not in used:
            used.append(words[i])
        else:
            answer.append(i % n + 1)
            answer.append(((i + 1 - answer[0]) // n) + 1)
            break

    if len(answer) == 0:
        answer.append(0)
        answer.append(0)
    return answer

File: 프로그래머스/javascript/LV2/영어 끝말잇기.js
function solution(n, words) {
  let target = -1;
  for (let i = 1; i < words.length; i++) {
    // 끝말잇기 실패
    if (words[i - 1].slice(-1) !== words[i][0]) {
      target = i;
      break;
    }
    // 중복된 단어
    if (words.indexOf(words[i]) !== i) {
      target = i;
      break;
    }
  }

  if (target === -1) return [0, 0];

  return [(target % n) + 1, Math.floor(target / n) + 1];
}
